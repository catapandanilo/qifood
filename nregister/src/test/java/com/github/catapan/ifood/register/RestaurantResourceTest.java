package com.github.catapan.ifood.register;

import static io.restassured.RestAssured.given;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import org.junit.jupiter.api.Test;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import javax.ws.rs.core.Response.Status;
import org.approvaltests.Approvals;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(RegisterTestLifecycleManager.class)
public class RestaurantResourceTest {

  @Test
  @DataSet("restaurants-scenery-1.yml")
  public void testGetRestaurants() {
    String result = given()
      .when().get("/restaurants")
      .then()
      .statusCode(Status.OK.getStatusCode())
      .extract().asString();
    Approvals.verifyJson(result);
  }
}
