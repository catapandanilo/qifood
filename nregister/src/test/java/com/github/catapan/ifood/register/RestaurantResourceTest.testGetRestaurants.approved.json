[
  {
    "id": 123,
    "name": "New Restaurant",
    "owner": "ID do keyclock"
  },
  {
    "id": 234,
    "name": "New Restaurant 2",
    "owner": "ID do keyclock 2"
  }
]